### PR TITLE
Fix array edit option

### LINKED ISSUE #8578: cannot edit array item
1. add field of type Array
2. add value item to that field
3. click edit and try edit the value 

bug: cannot edit the value

Image

### PR DESCRIPTION
**Pull request by BKM14**
Closes #8578
There was no case to handle a `FieldMetadataType.Array` and thus an error was thrown every time the edit button was clicked on an `Array` type.
It has been added now.

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/MultiItemFieldInput.tsx [hunk 1]
```diff
@@ -110,6 +110,7 @@ export const MultiItemFieldInput = <T,>({
         item = items[index] as PhoneRecord;
         setInputValue(item.countryCode + item.number);
         break;
+      case FieldMetadataType.Array:
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/MultiItemFieldInput.tsx:113]: Why is there no break condition for this? is the break condition for emails and arrays the same?

**Review by ehconitin — COMMENTED** [review]
Left a comment

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/MultiItemFieldInput.tsx (hunk 1 above)
Comment (BKM14) [reply packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/MultiItemFieldInput.tsx:113]: Yes, the break statement for Array and Email are the same. The way the Array is being handled in the backend is just like the Email. So I found it better to use the same block as the Email one

**Comment by BKM14** [comment]
Hey @ehconitin, could you review this please?

**Review by ehconitin — COMMENTED** [review]
This works, but I think we should have an explicit break for FieldMetadataType.Array. Having an explicit break makes it immediately clear that this is intentional behavior, not an oversight.

**Comment by BKM14** [comment]
There is an explicit break for the Array though. If it applies for Email, won't it also apply for Array

**Comment by ehconitin** [comment]
To clarify - currently, there isn't a separate break for Array. The code falls through from the Array case to the Email case, sharing a single break statement. While they handle the input the same way now, having separate cases with explicit breaks would make the code's intention clearer and easier to maintain.

What if in the future email break conditions change??? Your code will break in that situation :)

**Pull request closed without merging**